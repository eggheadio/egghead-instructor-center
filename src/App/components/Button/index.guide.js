import React from 'react'
import {storiesOf} from '@kadira/storybook'
import {lorem} from 'faker'
import Button from '.'

storiesOf('Button', module)
  .add('default', () => (
    <Button> 
      {lorem.words()}
    </Button>
  ))
