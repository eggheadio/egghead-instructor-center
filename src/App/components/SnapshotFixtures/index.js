import React from 'react'

export const StringFixture = 'String'

export const ImageUrlFixture = 'https://s3.amazonaws.com/uifaces/faces/twitter/kimcool/128.jpg'

export const ComponentFixture = () => (
  <div>
    Component
  </div>
)
