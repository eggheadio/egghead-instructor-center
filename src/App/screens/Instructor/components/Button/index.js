import React from 'react'

export default ({children, onClick, className}) => (
  <button
    className={`button-reset bn dib b0 ttu dim br2 ph3 pv2 white bg-orange tc ${className}`}
    onClick={onClick}
  >
    {children}
  </button>
)
