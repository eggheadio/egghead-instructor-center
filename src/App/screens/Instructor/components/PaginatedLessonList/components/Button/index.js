import React from 'react'

export default ({children, onClick}) => (
  <div
    className='ttu dim br2 ph3 pv2 white bg-orange tc'
    onClick={onClick}
  >
    {children}
  </div>
)
