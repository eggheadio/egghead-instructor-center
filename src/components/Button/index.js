import React from 'react'

export default ({children, onClick, subtle}) => (
  <button
    className={`
      button-reset bn dib b0 ttu dim br2 ph3 pv2 tc
      white ${subtle ? 'bg-black-50' : 'bg-orange'}
    `}
    onClick={onClick}
  >
    {children}
  </button>
)
