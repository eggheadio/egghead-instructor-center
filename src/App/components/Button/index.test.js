import React from 'react'
import {create} from 'react-test-renderer'
import {StringFixture} from '../SnapshotFixtures'
import Button from '.'

test('default', () => (
  expect(create(
    <Button>
      {StringFixture}
    </Button>
  )).toMatchSnapshot()
))
