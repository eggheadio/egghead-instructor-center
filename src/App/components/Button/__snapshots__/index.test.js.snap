exports[`test default 1`] = `
<button
  className="button-reset bn dib b0 ttu dim br2 ph3 pv2 white bg-orange tc undefined"
  onClick={undefined}>
  String
</button>
`;
